### ISSUE TITLE
Support non-number floating point values

### ISSUE THREAD

**Issue by TWiStErRob** [google/TestParameterInjector#14]
For negative scenario testing and edge cases this would be really useful. Right now to test these we have to write a custom provider.

```java
@TestParameter({"NaN", "-Infinity", "+Infinity", "Infinity"}) float myFloat;
@TestParameter({"NaN", "-Infinity", "+Infinity", "Infinity"}) double myDouble;
```

```
Could not map YAML value NaN (class = class java.lang.String) to java class float
java.lang.IllegalArgumentException: Could not map YAML value NaN (class = class java.lang.String) to java class float
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:453)
	at com.google.testing.junit.testparameterinjector.ParameterValueParsing$YamlValueTransformer.transformedJavaValue(ParameterValueParsing.java:187)
	at com.google.testing.junit.testparameterinjector.ParameterValueParsing.parseYamlObjectToJavaType(ParameterValueParsing.java:146)
	at com.google.testing.junit.testparameterinjector.ParameterValueParsing.parseYamlStringToJavaType(ParameterValueParsing.java:59)
	at com.google.testing.junit.testparameterinjector.TestParameter$InternalImplementationOfThisParameter.parseStringValue(TestParameter.java:194)
```

**Comment by nymanjens** [comment]
Thanks for raising this issue! Fixed in 6e66e93.

**Comment by TWiStErRob** [comment]
Nice, thank you!
Note: you forgot to add float tests.
